### DISCUSSION TITLE
PaddleOCR/main/algorithm/text_recognition/algorithm_rec_cppd

### DISCUSSION THREAD

**Discussion by giscus** [PaddlePaddle/PaddleOCR#14361] [Q&A]
# PaddleOCR/main/algorithm/text_recognition/algorithm_rec_cppd

Awesome multilingual OCR toolkits based on PaddlePaddle (practical ultra lightweight OCR system, support 80+ languages recognition, provide data annotation and synthesis tools, support training and deployment among server, mobile, embedded and IoT devices)

https://paddlepaddle.github.io/PaddleOCR/main/algorithm/text_recognition/algorithm_rec_cppd.html

**Comment by Dien666** [comment]
你好，在CPPD的推理模型部署章节，提供的模型下载链接是无效的哦。能否更正一下，或是评论区提供一下正确的。

**Reply by GreatV** [reply]
已修复

**Comment by Dien666** [comment]
请问一下使用这个初始化OCR时，是否可以随意替换检测和识别模型呢？
ocr = PaddleOCR(
    det_model_dir='./inference/ch_PP-OCRv4_det_server_infer',
    rec_model_dir='./inference/rec_svtr_cppd_base_en_infer',
    cls_model_dir='./inference/ch_PP-OCRv4_cls_infer',
    use_angle_cls=True, lang="ch"
)
我将rec_model_dir替换为rec_svtr_cppd_base_en_infer推理模型地址，但是运行有报错：
Traceback (most recent call last):
  File "D:\Users\bhgui\PycharmProjects\Odin\odin_app\PaddleDemo.py", line 16, in <module>
    result = ocr.ocr(img_path, cls=False)
  File "D:\Users\bhgui\PycharmProjects\Odin\.venv\lib\site-packages\paddleocr\paddleocr.py", line 760, in ocr
    dt_boxes, rec_res, _ = self.__call__(img, cls, slice)
  File "D:\Users\bhgui\PycharmProjects\Odin\.venv\lib\site-packages\paddleocr\tools\infer\predict_system.py", line 144, in __call__
    rec_res, elapse = self.text_recognizer(img_crop_list)
  File "D:\Users\bhgui\PycharmProjects\Odin\.venv\lib\site-packages\paddleocr\tools\infer\predict_rec.py", line 793, in __call__
    self.predictor.run()
ValueError: (InvalidArgument) Broadcast dimension mismatch. Operands could not be broadcast together with the shape of X = [4, 5064, 128] and the shape of Y = [1, 200, 128]. Received [5064] in X is not equal to [200] in Y at i:1.
  [Hint: Expected x_dims_array[i] == y_dims_array[i] || x_dims_array[i] <= 1 || y_dims_array[i] <= 1 == true, but received x_dims_array[i] == y_dims_array[i] || x_dims_array[i] <= 1 || y_dims_array[i] <= 1:0 != true:1.] (at ..\paddle/phi/kernels/funcs/common_shape.h:86)
  [operator < elementwise_add > error]

是否有章节讲如何替换模型呢？还是这种方式使用不支持？

**Reply by GreatV** [reply]
python api只有ppocr系列模型支持
